app/models/general.js

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export default class General {
+    get configAsJson() {
+        try {
+            return JSON.parse(this.config);
+        } catch {
+            return null;
+        }
+    }
+
+    set configFromJson(config) {
+        this.config = JSON.stringify(config);
+    }
+
+    get licenseAsJson() {
+        try {
+            return JSON.parse(this.license);
+        } catch {
+            return null;
+        }
+    }
+
+    set licenseFromJson(license) {
+        this.license = JSON.stringify(license);
+    }
+
+    get dataRetentionPolicyAsJson() {
+        try {
+            return JSON.parse(this.policy);
+        } catch {
+            return null;
+        }
+    }
+
+    set dataRetentionPolicyFromJson(policy) {
+        this.policy = JSON.stringify(policy);
+    }
+
+    static schema = {
+        name: 'General',
+        primaryKey: 'id',
+        properties: {
+            id: 'string', // make this id a constant in code
+            currentChannel: 'Channels?',
+            currentTeam: 'Teams?',
+            currentUser: 'Users?',
+            dataRetentionPolicy: 'string?',
+            deviceToken: 'string?',
+            serverConfig: 'string?',
+            serverLicense: 'string?',
+            serverVersion: 'string?',
+            timezones: 'string[]',
```

Comment:
Do the timezones need to live in Realm? I don't think they change at all, do they?

Reply:
good catch!

Reply:
Do they need to be persisted anywhere? I thought they would be constants

Reply:
We get them from the server

Reply:
Huh, that's weird. I guess that goes back to how you used to be able to (or maybe still can) limit the available time zones for whatever reason